eox_core/api/support/v1/tests/integration/test_views.py

```diff
@@ -0,0 +1,298 @@
+
+        Args:
+            tenant (dict): The tenant data.
+            params (dict, optional): The query parameters for the request.
+            data (dict, optional): The body data for the request.
+
+        Returns:
+            requests.Response: The response object.
+        """
+        return make_request(tenant, "PATCH", url=self.UPDATE_USERNAME_URL, params=params, data=data)
+
+    def create_oauth_application(self, tenant: dict, data: dict | None = None) -> requests.Response:
+        """
+        Create an oauth application in a tenant.
+
+        Args:
+            tenant (dict): The tenant data.
+            data (dict, optional): The body data for the request.
+
+        Returns:
+            requests.Response: The response object.
+        """
+        return make_request(tenant, "POST", url=self.OAUTH_APP_URL, json=data)
+
+
+@ddt.ddt
+class TestEdxAppUserAPIIntegration(
+    SupportAPIRequestMixin,
+    BaseIntegrationTest,
+    UsersAPIRequestMixin,
+):
+    """Integration tests for the EdxApp User API."""
+
+    @ddt.data("username", "email")
+    def test_delete_user_in_tenant_success(self, query_param: str) -> None:
+        """
+        Test delete an edxapp user in a tenant.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+        - `delete_edxapp_user`
+
+        Expected result:
+        - The status code is 200.
+        - The response indicates the user was removed successfully from the tenant.
+        - The user is not found in the tenant.
+        """
+        data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, data)
+
+        response = self.delete_user(self.tenant_x, {query_param: data[query_param]})
+        response_data = response.json()
+        get_response = self.get_user(self.tenant_x, {"email": data["email"]})
+        get_response_data = get_response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response_data, f"The user {data['username']} <{data['email']}>  has been removed")
+        self.assertEqual(get_response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(
+            get_response_data,
+            {"detail": f"No user found by {{'email': '{data['email']}'}} on site {self.tenant_x['domain']}."},
+        )
+
+    @ddt.data(
+        ("username", "user-not-found"),
+        ("email", "user-not-found@mail.com"),
+    )
+    @ddt.unpack
+    def test_delete_user_in_tenant_not_found(self, query_param: str, value: str) -> None:
+        """
+        Test delete an edxapp user in a tenant that does not exist.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+
+        Expected result:
+        - The status code is 404.
+        - The response indicates the user was not found in the tenant.
+        """
+        response = self.delete_user(self.tenant_x, {query_param: value})
+        response_data = response.json()
```

Comment:
What if I try to delete a user from a different tenant? Should that work?

Reply:
I added a test for this: https://github.com/eduNEXT/eox-core/commit/47e03c993866e0d63471354a2d0b1f2469e1aa71